src/api/interface.ts

```diff
@@ -21,24 +21,50 @@ enum ResolveError {
 /**
 * Intent descriptor
 */
-interface IntentMetadata {
+interface Intent {
+  
+  // The name of the intent, default naming convention is UpperCamelCase
   name: string;
-  displayName: string;
+
+  // An optional display name for the intent, "name" will be used if left empty
+  displayName?: string;
+  
+  // A list of the contexts the intent accepts. This will typically be a set of namespaced context types, e.g. "fdc3.contact"
+  contexts?: string[]
+ 
+  // Custom configuration for the intent that may be required for a particular desktop agent.
+  customConfig?: object; 
 }
 
 /**
 * An interface that relates an intent to apps
 */
 interface AppIntent {
-  intent: IntentMetadata;
-  apps: Array<AppMetadata>;
+  intent: Intent;
+  apps: Array<Application>;
 }
 
 /**
- * App metadata is Desktop Agent specific - but should support a name property.
+ * App definition as provided by the application directory
  */
-interface AppMetadata {
+interface Application {
+  appId: string;
   name: string;
```

Comment:
What is the difference and intended usage of `name` vs `appId` (the AppD schema requires both of them to be unique). Which one is supposed to be used with `raiseIntent`'s `target` parameter?